src/main/java/com/elixr/poc/rest/controller/PurchaseDeletionController.java

```diff
@@ -1,50 +1,45 @@
-//    private final PurchaseService purchaseService;
-//
-//    public PurchaseDeletionController(PurchaseService purchaseService) {
-//        this.purchaseService = purchaseService;
-//    }
-//
-//    /**
-//     * Calling deletePurchaseDetails method with the parameter of the purchaseId to delete purchase and
-//     * handling the exception if Id is not found
-//     *
-//     * @param purchaseId
-//     * @return
-//     */
-//    @DeleteMapping("/purchase/{purchaseId}")
-//    public ResponseEntity<?> deletePurchase(@PathVariable("purchaseId") UUID purchaseId) {
-//        DeleteResponse deleteResponse = new DeleteResponse();
-//        HttpStatus httpStatus;
-//        try {
-//            boolean success = purchaseService.deletePurchaseDetails(purchaseId);
-//            deleteResponse.setSuccess(success);
-//            deleteResponse.setErrorMessages(ApplicationConstants.DELETED);
-//            httpStatus = HttpStatus.OK;
-//        } catch (NoRecordFoundException e) {
-//            deleteResponse.setSuccess(false);
-//            deleteResponse.setErrorMessages(e.getMessage());
-//            httpStatus = HttpStatus.BAD_REQUEST;
-//        }
-//        return new ResponseEntity<>(deleteResponse, httpStatus);
-//    }
-//}
+package com.elixr.poc.rest.controller;
+
+import com.elixr.poc.common.MessagesKeyEnum;
+import com.elixr.poc.common.util.MessagesUtil;
+import com.elixr.poc.exception.IdNotFoundException;
+import com.elixr.poc.rest.response.CommonResponse;
+import com.elixr.poc.service.PurchaseService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/controller")
+/**
+ * purchaseDeletionController is invoking the purchaseService.
+ */ public class PurchaseDeletionController {
+    private final PurchaseService purchaseService;
+
+    public PurchaseDeletionController(PurchaseService purchaseService) {
+        this.purchaseService = purchaseService;
+    }
+
+    /**
+     * Calling deletePurchaseDetails method with the parameter of the purchaseId to delete purchase and
+     * handling the exception if Id is not found
+     *
+     * @param purchaseId
+     * @return
+     */
+    @DeleteMapping("/purchase/{purchaseId}")
+    public ResponseEntity deletePurchase(@PathVariable("purchaseId") UUID purchaseId) throws IdNotFoundException {
+        CommonResponse deleteResponse;
+        HttpStatus httpStatus;
+        boolean success = purchaseService.deletePurchaseDetails(purchaseId);
+        deleteResponse = CommonResponse.builder().success(success)
+                .errorMessage(MessagesUtil.getMessage(MessagesKeyEnum.ENTITY_DELETED_SUCCESSFULLY.getKey())).build();
```

Comment:
Why are you passing success message in the 'errorMessage' field?

Reply:
Updated